src/schema/models/schemaManager.ts

```diff
@@ -0,0 +1,48 @@
+import _ from 'lodash';
+import { inject, injectable } from 'tsyringe';
+import { Schemas, Schema, Tags } from './mapping';
+
+@injectable()
+export class SchemaManager {
+  public constructor(@inject(Schemas) private readonly schemas: Schemas) {}
+
+  public async getSchemas(): Promise<Schema[]> {
+    return this.schemas.getSchemas();
+  }
+
+  public async getSchema(name: string): Promise<Schema | undefined> {
+    return this.schemas.getSchema(name);
+  }
+
+  public async map(name: string, tags: Tags, schema: Schema): Promise<Tags> {
+    let tagsCopy = { ...tags };
+
+    tagsCopy = await this.mapTags(tags, schema);
+    tagsCopy = await this.addSystemName(tagsCopy, name);
```

Comment:
right now we only add system name as a prefix, that is the current mapping, not need to `mapTags` and `addSystemName`

also `addSystemName` is not an informative function name, one can ask 
> addSystemName to what?

adding a system name as a prefix is the current strategy for mapping tags => current implementation of `mapTags` should be `addSystemName`

Reply:
Done